modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PlatformPreferences.java

```diff
@@ -0,0 +1,238 @@
+        invalidationListeners.remove(listener);
+    }
+
+    @Override
+    public void addListener(MapChangeListener<? super String, ? super Object> listener) {
+        mapChangeListeners.add(listener);
+    }
+
+    @Override
+    public void removeListener(MapChangeListener<? super String, ? super Object> listener) {
+        mapChangeListeners.remove(listener);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Optional<T> getValue(String key, Class<T> type) {
+        Objects.requireNonNull(key, "key cannot be null");
+        Objects.requireNonNull(key, "type cannot be null");
+        Object value = effectivePreferences.get(key);
+
+        if (value == null) {
+            return Optional.empty();
+        }
+
+        if (type.isInstance(value)) {
+            return Optional.of((T)value);
+        }
+
+        throw new IllegalArgumentException(
+            "Incompatible types: requested = " + type.getName() +
+            ", actual = " + value.getClass().getName());
+    }
+
+    @Override
+    public Optional<Integer> getInteger(String key) {
+        return getValue(key, Integer.class);
+    }
+
+    @Override
+    public Optional<Double> getDouble(String key) {
+        return getValue(key, Double.class);
+    }
+
+    @Override
+    public Optional<Boolean> getBoolean(String key) {
+        return getValue(key, Boolean.class);
+    }
+
+    @Override
+    public Optional<String> getString(String key) {
+        return getValue(key, String.class);
+    }
+
+    @Override
+    public Optional<Color> getColor(String key) {
+        return getValue(key, Color.class);
+    }
+
+    @Override
+    public Optional<Paint> getPaint(String key) {
+        return getValue(key, Paint.class);
+    }
+
+    @Override
+    public ReadOnlyObjectProperty<Appearance> appearanceProperty() {
+        return properties.appearanceProperty();
+    }
+
+    @Override
+    public Appearance getAppearance() {
+        return properties.getAppearance();
+    }
+
+    @Override
+    public ReadOnlyObjectProperty<Color> backgroundColorProperty() {
+        return properties.backgroundColorProperty();
+    }
+
+    @Override
+    public Color getBackgroundColor() {
+        return properties.getBackgroundColor();
+    }
+
+    @Override
+    public ReadOnlyObjectProperty<Color> foregroundColorProperty() {
+        return properties.foregroundColorProperty();
+    }
+
+    @Override
+    public Color getForegroundColor() {
+        return properties.getForegroundColor();
+    }
+
+    @Override
+    public ReadOnlyObjectProperty<Color> accentColorProperty() {
+        return properties.accentColorProperty();
+    }
+
+    @Override
+    public Color getAccentColor() {
+        return properties.getAccentColor();
+    }
+
+    /**
+     * Updates this map of preferences with a new set of platform preferences.
+     * The specified preferences may include all available preferences, or only the changed preferences.
```

Comment:
I would mention here that removed preferences are specified by being mapped to `null`, but that the resulting preferences (after update) cannot map to `null`.

Reply:
Done.